nixops/backends/libvirtd.py

```diff
@@ -217,10 +224,15 @@ def read_file(stream, nbytes, f):
             stream.sendAll(read_file, f)
             stream.finish()
 
+    def _qemu_executable(self):
+        domaincaps_xml = self.conn.getDomainCapabilities(
+            emulatorbin=None, arch='x86_64', machine=None, virttype='kvm',
+        )
+        domaincaps = ElementTree.fromstring(domaincaps_xml)
+        return domaincaps.find('./path').text.strip()
+
     def _make_domain_xml(self, defn):
-        qemu_executable = "qemu-system-x86_64"
-        qemu = spawn.find_executable(qemu_executable)
-        assert qemu is not None, "{} executable not found. Please install QEMU first.".format(qemu_executable)
+        qemu = self._qemu_executable()
```

Comment:
wouldn't it be better to keep the assert ? I would miss the error message.

Reply:
You are right, if QEMU is not installed, the call to `libvirt.open()` will fail with some unclear exception like `no connection driver available for qemu:///system`. I'll try to handle the error and output some useful message to the user.